app/src/org/commcare/dalvik/preferences/DeveloperPreferences.java

```diff
@@ -43,27 +42,123 @@
+    public final static String ANALYTICS_ENABLED = "cc-analytics-enabled";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        PreferenceManager prefMgr = getPreferenceManager();
+        GoogleAnalyticsUtils.reportPrefActivityEntry(GoogleAnalyticsFields.CATEGORY_DEV_PREFS);
 
+        PreferenceManager prefMgr = getPreferenceManager();
         prefMgr.setSharedPreferencesName((CommCareApplication._().getCurrentApp().getPreferencesFilename()));
 
         addPreferencesFromResource(R.xml.preferences_developer);
         setTitle("Developer Preferences");
+
+        createPreferenceOnClickListeners(prefMgr);
+    }
+
+    private void createPreferenceOnClickListeners(PreferenceManager prefManager) {
+        String[] prefKeys = {
+                SUPERUSER_ENABLED,
+                ACTION_BAR_ENABLED,
+                GRID_MENUS_ENABLED,
+                NAV_UI_ENABLED,
+                LIST_REFRESH_ENABLED,
+                NEWEST_APP_VERSION_ENABLED,
+                ENABLE_AUTO_LOGIN,
+                ENABLE_SAVE_SESSION,
+                CSS_ENABLED,
+                MARKDOWN_ENABLED,
+                ALTERNATE_QUESTION_LAYOUT_ENABLED,
+                FIRE_TRIGGERS_ON_SAVE,
+                HOME_REPORT_ENABLED};
+        String[] analyticsLabels = {
+                GoogleAnalyticsFields.LABEL_DEV_MODE,
+                GoogleAnalyticsFields.LABEL_ACTION_BAR,
+                GoogleAnalyticsFields.LABEL_GRID_MENUS,
+                GoogleAnalyticsFields.LABEL_NAV_UI,
+                GoogleAnalyticsFields.LABEL_ENTITY_LIST_REFRESH,
+                GoogleAnalyticsFields.LABEL_NEWEST_APP_VERSION,
+                GoogleAnalyticsFields.LABEL_AUTO_LOGIN,
+                GoogleAnalyticsFields.LABEL_SESSION_SAVING,
+                GoogleAnalyticsFields.LABEL_CSS,
+                GoogleAnalyticsFields.LABEL_MARKDOWN,
+                GoogleAnalyticsFields.LABEL_IMAGE_ABOVE_TEXT,
+                GoogleAnalyticsFields.LABEL_TRIGGERS_ON_SAVE,
+                GoogleAnalyticsFields.LABEL_REPORT_BUTTON_ENABLED};
+
+        for (int i = 0; i < prefKeys.length; i++) {
+            GoogleAnalyticsUtils.createPreferenceOnClickListener(prefManager, prefKeys[i],
+                    GoogleAnalyticsFields.CATEGORY_DEV_PREFS, analyticsLabels[i]);
+        }
     }
 
     @Override
     public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
-        if (key.equals(ENABLE_AUTO_LOGIN) && !isAutoLoginEnabled()) {
-            DevSessionRestorer.clearPassword(sharedPreferences);
-        } else if (key.equals(ENABLE_SAVE_SESSION) && !isSessionSavingEnabled()) {
-            DevSessionRestorer.clearSession();
+        String editPrefLabel = "";
+        switch(key) {
+            case SUPERUSER_ENABLED:
+                editPrefLabel = GoogleAnalyticsFields.LABEL_DEV_MODE;
```

Comment:
I'm confused, we build a big table of these above that we should be able to use to create this table, right? I don't think we should have to manually code it in again?